src/onnx/parse_resize.cpp

```diff
@@ -21,102 +21,82 @@
-     append to vec_ind.
- * Inner loop:
- *   Compose a vector of indices by iterating all layer-1 using current bitset from current element.
- *
- * i.e. val = 6 -> bitset 0110b -> indices: pick each value from A'D'F'G' -> in_s.index(indices)
+ * Output: per resized pixel, its neighboring hi/lo indexes (vector): all permutations.
+ * This api stitches all the neighbors (for every dimension) for a resized pixel,
+ * to yield its neighbor index w.r.t to the input shape, in_s.
  */
 
 static std::vector<int>
 calc_neighbor_points(const std::vector<std::vector<std::vector<std::size_t>>>& vvv_ind,
-                     const shape& in_s)
+                     const shape& in_s,
+                     const shape& out_s,
+                     const std::map<size_t, size_t>& resized_m)
 {
-    std::size_t n_bits     = vvv_ind.size();
-    std::size_t m_elements = vvv_ind[0][0].size();
-    std::vector<int> vec_ind;
+    std::size_t ndims       = out_s.ndim();
+    const auto& strides     = out_s.strides();
+    std::size_t elements_ct = vvv_ind[0][0].size();
 
-    if(n_bits >= std::numeric_limits<std::size_t>::digits)
-    {
-        MIGRAPHX_THROW("PARSE_RESIZE: Shape dimension " + std::to_string(n_bits) + " exceeds " +
-                       std::to_string(std::numeric_limits<std::size_t>::digits));
-    }
+    // This function computes for each element, all permutations of its neighbor indices into an
+    // Perm block in one go. (Instead of computing each permutation in isolation per element)
+    size_t permutations = 1u << resized_m.size();
+    std::vector<std::vector<std::size_t>> perm_blk(permutations, std::vector<size_t>(strides));
 
-    for(std::size_t val = 0; val < (std::size_t{1} << n_bits); val++)
+    // final outputted vector: permutations of neighbors.
+    std::vector<int> out_idx_vec(permutations * elements_ct);
+
+    for(size_t e_idx = 0; e_idx < elements_ct; ++e_idx)
     {
-        std::bitset<std::numeric_limits<std::size_t>::digits> bits_val = val;
-        std::vector<std::size_t> indices(n_bits);
-        transform(range(m_elements), std::back_inserter(vec_ind), [&](std::size_t i_element) {
-            transform(
-                vvv_ind, range(n_bits), indices.begin(), [&](const auto& vv_ind, std::size_t bit) {
-                    return vv_ind[bits_val[bit]][i_element];
-                });
-            return in_s.index(indices);
-        });
+        size_t t_idx = e_idx;
+        for(size_t l_idx = 0; l_idx != ndims; ++l_idx)
+        {
+            auto entry = resized_m.find(l_idx);
+            if(entry != resized_m.end())
+            {
+                size_t hi_cmp_bit = 1u << entry->second;
+                auto lo           = vvv_ind[entry->second][0][e_idx];
+                auto hi           = vvv_ind[entry->second][1][e_idx];
+                for(size_t i = 0; i < permutations; i++)
+                    perm_blk[i][l_idx] = ((i & hi_cmp_bit) != 0) ? hi : lo;
```

Comment:
What is this supposed to do? There is no explanation here. Using a bitset like the previous version would be better.